### PR TITLE
refactor(site): refactor external auth component

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Recommended improvements:
- Rename component for clarity 
- Simplify interface for contextual relevance 
- Handle polling errors based on section, not every button

Before:
Screenshot 2024-01-22 at 15 24 26

Now:
Screenshot 2024-01-22 at 15 24 41

### CODE REVIEW

File: site/src/pages/CreateWorkspacePage/CreateWorkspacePage.test.tsx [hunk 1]
```diff
@@ -201,25 +201,17 @@ describe("CreateWorkspacePage", () => {
...
     jest
       .spyOn(API, "getTemplateVersionExternalAuth")
       .mockResolvedValueOnce([MockTemplateVersionExternalAuthGithub]);
 
     renderCreateWorkspacePage();
     await waitForLoaderToBeRemoved();
 
-    const nameField = await screen.findByLabelText(nameLabelText);
-
-    // have to use fireEvent b/c userEvent isn't cleaning up properly between tests
```
Comment (Parkreiner) [inline site/src/pages/CreateWorkspacePage/CreateWorkspacePage.test.tsx:214]: I know that we have the test simplified down now, so it's not an issue anymore, but do you know if we ever figured out the `userEvent` issue? This is the first I've heard of it not cleaning up properly, but some digging made it sounds like it was a bug with the library

**Review by buenos-nachos — APPROVED** [review]
Code looks good, and big fan of the change!

File: site/src/pages/CreateWorkspacePage/ExternalAuthButton.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+export const ExternalAuthButton: FC<ExternalAuthButtonProps> = ({
+  auth,
+  displayRetry,
+  isLoading,
+  onStartPolling,
+}) => {
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (BrunoQuaresma) [reply site/src/pages/CreateWorkspacePage/ExternalAuthButton.tsx:24]: Good catch

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/11742

**Pull request merged**